[TS] LPS-91113 Avoid extra DB calls when portlets undeployed

marianoalvarosaiz opened the pull request.

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

topolik commented:
Hi @marianoalvarosaiz, can you please write me some info about this fix? I see your fixed the "count" method and then the fetch + strict but I'd like to understand better.

Thanks!

marianoalvarosaiz commented:
Hi @topolik,
 I followed your advices and:

1. I added a breakpoint to QueryPos and saw that those were the 3 methods involved.
2. I saw that when the strict method was called using settingsId as portetId it was for shared preferences (group or company), hence a plid equals to 0, and when used from p_p_id plid had a non shared value, that's why I added the additional condition in addtion to the first pull.
3. For the fetch/count I also added the plid condition to avoid non being able to retrieve the company from the plid, not trusting to much the CompanyThreadLocal to retrieve it.

Kind regards.

topolik commented on portal-impl/src/com/liferay/portal/service/impl/PortletPreferencesLocalServiceImpl.java:
```diff
@@ -365,8 +373,16 @@ public PortletPreferences getPortletPreferences(
 	public long getPortletPreferencesCount(
 		int ownerType, long plid, String portletId) {
 
+		plid = _swapPlidForPortletPreferences(plid);
```
Is this needed? Have you tried with staging enabled and layout revision?

marianoalvarosaiz replied:
Thanks a lot. I've tested it and the code we add should be before performing the swap, to be able to retrieve the plid (not the revision) to fetch the company.

topolik commented on portal-impl/src/com/liferay/portal/service/impl/PortletPreferencesLocalServiceImpl.java:
```diff
@@ -175,6 +175,14 @@ public void deletePortletPreferencesByPlid(long plid) {
 	public PortletPreferences fetchPortletPreferences(
 		long ownerId, int ownerType, long plid, String portletId) {
 
+		if (plid > 0) {
```
Is there a reason we use `0` here and `PortletKeys.PREFS_PLID_SHARED` on line 567?

marianoalvarosaiz replied:
I thought PREFS_PLID_SHARED was more like an enum value, and sounded not very intuitive comparing if PREFS_PLID_SHARED is bigger than 0. Although plid can have additional values for me PREFS_PLID_SHARED is more like a binary value, either it's shared or is not. Anyway it's only an opinion, so I'm ready to change it.

topolik commented on portal-impl/src/com/liferay/portal/service/impl/PortletPreferencesLocalServiceImpl.java:
```diff
@@ -631,6 +658,17 @@ public PortletPreferences updatePreferences(
+		return portletLocalService.fetchPortletById(
+			layout.getCompanyId(), portletId);
+	}
+
```
Looking at how we use it I think we could name it by our purpose, i.e. if portlet exists.

Actually because we always check the `plid != 0` all the checks could be encapsulated here with a method name that would express meaning of this fix. I'm bad at naming but it could be something like 'doesLayoutPortletExist()`? :)

marianoalvarosaiz replied:
That name sounds great.

topolik commented:
Thanks @marianoalvarosaiz for the answers, I understand now. I think this is the right way, I'd only make some small changes to better express the meaning of the `if` conditions, by placing them into a well named method. This way it will be easier to understand for others. 

Thanks.

topolik commented:
@marianoalvarosaiz waiting for your answers, no rush.

topolik commented:
ci:test

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest(master)/541/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-18.liferay.com//userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/541/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:x: ci:test - 161 out of 202 jobs passed in 3 hours 14 minutes 26 seconds 994 ms</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/19bbbf1ad18eff190403f1c452b1ece9eadf2ddd">19bbbf1ad18eff190403f1c452b1ece9eadf2ddd</a></p><h4>Copied in Private Modules Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal-ee/tree/master-private">master-private</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal-ee/commit/bb05b18168fd1ff84bb232f565de282ff8c473b8">bb05b18168fd1ff84bb232f565de282ff8c473b8</a></p>161 out of 202 jobs PASSED<div><h4>41 Failed Jobs:</h4><ul><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest(master)/541/"><strike><strong>test-portal-acceptance-pullrequest(master)</strong></strike></a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/282927/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-smoke-weblogic122-mysql57-jdk8/0</strong></strike></a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/148605/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-smoke-websphere90-mysql57-jdk8/0</strong></strike></a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/148606/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-smoke-wildfly110-mariadb102-jdk8/0</strong></strike></a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/275778/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/3</strong></strike></a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/238863/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/30</strong></strike></a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/283076/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/36</strong></strike></a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/252404/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/4</strong></strike></a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/274497/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/51</strong></strike></a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/225829/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/55</strong></strike></a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/274500/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tcserver40-mysql57-jdk8/2</strong></strike></a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/288042/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tcserver40-mysql57-jdk8/3</strong></strike></a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/148614/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tcserver40-mysql57-jdk8/4</strong></strike></a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/282936/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tcserver40-mysql57-jdk8/5</strong></strike></a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/299347/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-postgresql10-jdk8/2</strong></strike></a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/283083/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-weblogic122-mysql57-jdk8/2</strong></strike></a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/297110/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-weblogic122-mysql57-jdk8/3</strong></strike></a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/282939/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-websphere90-mysql57-jdk8/0</strong></strike></a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/274502/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-websphere90-mysql57-jdk8/1</strong></strike></a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/276920/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-websphere90-mysql57-jdk8/2</strong></strike></a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/244000/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-websphere90-mysql57-jdk8/3</strong></strike></a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/276921/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-websphere90-mysql57-jdk8/4</strong></strike></a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/292154/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-websphere90-mysql57-jdk8/5</strong></strike></a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/283084/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-wildfly110-mariadb102-jdk8/1</strong></strike></a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/149404/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/integration-db2111-jdk8</strong></strike></a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/36164/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/integration-hypersonic20-jdk8</strong></strike></a></li><li><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/194260/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/integration-mariadb102-jdk8</strong></strike></a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/148619/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/integration-mysql57-jdk8</strong></strike></a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/260965/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/integration-oracle122-jdk8</strong></strike></a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/276922/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/integration-postgresql10-jdk8</strong></strike></a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/292156/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/integration-sybase160-jdk8</strong></strike></a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/282940/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/modules-compile-jdk8</strong></strike></a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/299351/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/modules-integration-db2111-jdk8</strong></strike></a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/276923/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/modules-integration-hypersonic20-jdk8</strong></strike></a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/225834/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/modules-integration-mariadb102-jdk8</strong></strike></a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/149406/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8</strong></strike></a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/241765/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/modules-integration-oracle122-jdk8</strong></strike></a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/288046/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/modules-integration-postgresql10-jdk8</strong></strike></a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/148620/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/modules-integration-sybase160-jdk8</strong></strike></a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/244002/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/modules-semantic-versioning-jdk8</strong></strike></a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/292157/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8</strong></strike></a></li></ul></div><details><summary><strong>161 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/285123/">test-portal-acceptance-pullrequest-batch(master)/bundle-blacklist-restart-jdk8</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/285124/">test-portal-acceptance-pullrequest-batch(master)/central-requirements-jdk8</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/252399/">test-portal-acceptance-pullrequest-batch(master)/compile-jsp-jdk8</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/285125/">test-portal-acceptance-pullrequest-batch(master)/deprecated-module-smoke-jdk8</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/276907/">test-portal-acceptance-pullrequest-batch(master)/empty-temp-dir-jdk8</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/252400/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-jboss71-mysql57-jdk8/0</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/252401/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tcserver40-mysql57-jdk8/0</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/252402/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-db2111-jdk8/0</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/292150/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mariadb102-jdk8/0</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/274490/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk11_open/0</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/276908/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/148603/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-oracle122-jdk8/0</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/276909/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-postgresql10-jdk8/0</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/148604/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-sybase160-jdk8/0</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/282928/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-db2111-jdk8/0</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/276910/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-db2111-jdk8/1</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/282929/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-db2111-jdk8/2</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/282930/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-hypersonic20-jdk8/0</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/274491/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-hypersonic20-jdk8/1</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/276911/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-hypersonic20-jdk8/2</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/148608/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-hypersonic20-jdk8/3</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/225822/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mariadb102-jdk8/0</a></li><li><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/194253/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mariadb102-jdk8/1</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/148609/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mariadb102-jdk8/2</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/252403/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/241761/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/1</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/297097/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/10</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/274494/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/11</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/275780/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/12</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/285129/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/13</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/261829/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/14</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/292152/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/15</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/241763/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/16</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/252405/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/17</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/238862/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/18</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/148611/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/19</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/285127/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/2</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/297098/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/20</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/275781/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/21</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/276913/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/22</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/297099/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/23</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/261830/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/24</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/274495/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/25</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/241764/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/26</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/285130/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/27</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/252406/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/28</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/282933/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/29</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/148612/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/31</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/292153/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/32</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/276914/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/33</a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/283075/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/34</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/225825/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/35</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/225826/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/37</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/297100/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/38</a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/283077/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/39</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/261831/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/40</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/225827/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/41</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/297101/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/42</a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/283078/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/43</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/274496/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/44</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/276915/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/45</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/261832/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/46</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/225828/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/47</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/282934/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/48</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/288039/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/49</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/292151/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/5</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/261833/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/50</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/276916/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/52</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/297102/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/53</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/282935/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/54</a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/283079/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/56</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/274498/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/57</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/299344/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/58</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/261828/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/6</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/241762/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/7</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/275779/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/8</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/285128/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/9</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/288040/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-oracle122-jdk8/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/288041/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-oracle122-jdk8/1</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/297103/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-oracle122-jdk8/2</a></li><li><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/194255/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-postgresql10-jdk8/0</a></li><li><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/194256/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-postgresql10-jdk8/1</a></li><li><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/194257/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-postgresql10-jdk8/2</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/275782/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-sybase160-jdk8/0</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/149402/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-sybase160-jdk8/1</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/297104/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-sybase160-jdk8/2</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/297105/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-jboss71-mysql57-jdk8/0</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/238864/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-jboss71-mysql57-jdk8/1</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/238865/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-jboss71-mysql57-jdk8/2</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/297106/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-jboss71-mysql57-jdk8/3</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/275783/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-jboss71-mysql57-jdk8/4</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/148613/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-jboss71-mysql57-jdk8/5</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/274499/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tcserver40-mysql57-jdk8/0</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/276917/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tcserver40-mysql57-jdk8/1</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/261834/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-db2111-jdk8/0</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/299345/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-db2111-jdk8/1</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/285131/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-db2111-jdk8/2</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/288043/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-db2111-jdk8/3</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/275784/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mariadb102-jdk8/0</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/275785/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mariadb102-jdk8/1</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/275786/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mariadb102-jdk8/2</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/297107/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mariadb102-jdk8/3</a></li><li><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/194258/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mariadb102-jdk8/4</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/243999/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mariadb102-jdk8/5</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/148615/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/297108/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/1</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/238866/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/2</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/148616/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/3</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/297109/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/4</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/225830/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/5</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/282937/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/6</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/238867/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/7</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/149403/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-oracle122-jdk8/0</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/282938/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-oracle122-jdk8/1</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/276918/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-oracle122-jdk8/2</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/276919/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-oracle122-jdk8/3</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/299346/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-postgresql10-jdk8/0</a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/283080/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-postgresql10-jdk8/1</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/225831/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-postgresql10-jdk8/3</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/288044/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-postgresql10-jdk8/4</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/274501/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-postgresql10-jdk8/5</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/275787/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-sybase160-jdk8/0</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/275788/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-sybase160-jdk8/1</a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/283081/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-sybase160-jdk8/2</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/285132/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-sybase160-jdk8/3</a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/283082/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-weblogic122-mysql57-jdk8/0</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/148617/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-weblogic122-mysql57-jdk8/1</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/225832/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-weblogic122-mysql57-jdk8/4</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/148618/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-weblogic122-mysql57-jdk8/5</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/299348/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-wildfly110-mariadb102-jdk8/0</a></li><li><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/194259/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-wildfly110-mariadb102-jdk8/2</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/299349/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-wildfly110-mariadb102-jdk8/3</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/260964/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-wildfly110-mariadb102-jdk8/4</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/292155/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-wildfly110-mariadb102-jdk8/5</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/149405/">test-portal-acceptance-pullrequest-batch(master)/javadoc-jdk8</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/244001/">test-portal-acceptance-pullrequest-batch(master)/js-unit-jdk8</a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/283085/">test-portal-acceptance-pullrequest-batch(master)/jsp-runtime-compile-jdk8</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/299350/">test-portal-acceptance-pullrequest-batch(master)/lpkg-base-jdk8</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/275789/">test-portal-acceptance-pullrequest-batch(master)/lpkg-container-jdk8</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/36165/">test-portal-acceptance-pullrequest-batch(master)/lpkg-controller-jdk8</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/288045/">test-portal-acceptance-pullrequest-batch(master)/lpkg-independence-jdk8</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/225833/">test-portal-acceptance-pullrequest-batch(master)/lpkg-override-jdk8</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/252407/">test-portal-acceptance-pullrequest-batch(master)/lpkg-persistence-jdk8</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/260966/">test-portal-acceptance-pullrequest-batch(master)/lpkg-release-jdk8</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/275790/">test-portal-acceptance-pullrequest-batch(master)/lpkg-startup-jdk8</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/274503/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-project-templates-jdk8</a></li><li><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/194261/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/297111/">test-portal-acceptance-pullrequest-batch(master)/portal-license-jdk8</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/285133/">test-portal-acceptance-pullrequest-batch(master)/portal-second-startup-space-in-path-jdk8</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/285134/">test-portal-acceptance-pullrequest-batch(master)/portal-startup-space-in-path-jdk8</a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/283086/">test-portal-acceptance-pullrequest-batch(master)/portal-web-jdk8</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/297112/">test-portal-acceptance-pullrequest-batch(master)/release-osgi-state-exploded-test-jdk8</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/260967/">test-portal-acceptance-pullrequest-batch(master)/release-osgi-state-lpkg-test-jdk8</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/282941/">test-portal-acceptance-pullrequest-batch(master)/ruby-sass-compiler-jdk8</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/261835/">test-portal-acceptance-pullrequest-batch(master)/semantic-versioning-jdk8</a></li><li><a href="https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/285135/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/260968/">test-portal-acceptance-pullrequest-batch(master)/slim-profile-jdk8</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/274504/">test-portal-acceptance-pullrequest-batch(master)/source-format-jdk8</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/252408/">test-portal-acceptance-pullrequest-batch(master)/tck-jdk8</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/297113/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/275791/">test-portal-acceptance-pullrequest-batch(master)/wsdd-builder-jdk8</a></li></ul></details><h5>For more details click <a href="https://test-1-18.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/541/jenkins-report.html">here</a>.</h5><hr/><h4>Failures unique to this pull:</h4><ol><li><div><h5><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/148606/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-wildfly110-mariadb102-jdk8/0</a></h5><div><h6>Job Results:</h6><p>3 Tests Passed.<br/>1 Test Failed.</p></div><ol><li><div><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/148606//consoleText">AXIS_VARIABLE=0,label_exp=!master #148606</a><ol><li><div><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/148606//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_PortalSmoke_Smoke_">LocalFile.PortalSmoke#Smoke</a> - <a href="https://testray.liferay.com/reports/production/logs/test-1-18/1551958464503/test-portal-acceptance-pullrequest(master)/541/functional-smoke-wildfly110-mariadb102-jdk8/0/0/LocalFile.PortalSmoke_Smoke/index.html.gz">Poshi Report</a> - <a href="https://testray.liferay.com/reports/production/logs/test-1-18/1551958464503/test-portal-acceptance-pullrequest(master)/541/functional-smoke-wildfly110-mariadb102-jdk8/0/0/LocalFile.PortalSmoke_Smoke/summary.html.gz">Poshi Summary</a> - <a href="https://testray.liferay.com/reports/production/logs/test-1-18/1551958464503/test-portal-acceptance-pullrequest(master)/541/functional-smoke-wildfly110-mariadb102-jdk8/0/0/jenkins-console.txt.gz">Console Output</a></div></li></ol></div></li></ol></div></li><li><div><h5><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/238863/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/30</a></h5><div><h6>Job Results:</h6><p>23 Tests Passed.<br/>1 Test Failed.</p></div><ol><li><div><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=1,label_exp=!master/238863//consoleText">AXIS_VARIABLE=1,label_exp=!master #238863</a><ol><li><div><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=1,label_exp=!master/238863//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_ExportImport_ExportImportPortlet_">LocalFile.ExportImport#ExportImportPortlet</a> - <a href="https://testray.liferay.com/reports/production/logs/test-1-18/1551958464503/test-portal-acceptance-pullrequest(master)/541/functional-tomcat90-mysql57-jdk8/30/1/LocalFile.ExportImport_ExportImportPortlet/index.html.gz">Poshi Report</a> - <a href="https://testray.liferay.com/reports/production/logs/test-1-18/1551958464503/test-portal-acceptance-pullrequest(master)/541/functional-tomcat90-mysql57-jdk8/30/1/LocalFile.ExportImport_ExportImportPortlet/summary.html.gz">Poshi Summary</a> - <a href="https://testray.liferay.com/reports/production/logs/test-1-18/1551958464503/test-portal-acceptance-pullrequest(master)/541/functional-tomcat90-mysql57-jdk8/30/1/jenkins-console.txt.gz">Console Output</a></div></li></ol></div></li></ol></div></li><li><div><h5><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/274497/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/51</a></h5><div><h6>Job Results:</h6><p>0 Tests Passed.<br/>0 Tests Failed.</p></div><ol><li><div><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/274497//consoleText">AXIS_VARIABLE=0,label_exp=!master #274497</a><pre><code>Build was aborted</code></pre></div></li></ol></div></li><li><div><h5><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/274500/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tcserver40-mysql57-jdk8/2</a></h5><div><h6>Job Results:</h6><p>1 Test Passed.<br/>1 Test Failed.</p></div><ol><li><div><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/274500//consoleText">AXIS_VARIABLE=0,label_exp=!master #274500</a><ol><li><div><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/274500//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_PortalSmokeUpgrade_ViewPortalSmokeArchive625_">LocalFile.PortalSmokeUpgrade#ViewPortalSmokeArchive625</a> - <a href="https://testray.liferay.com/reports/production/logs/test-1-18/1551958464503/test-portal-acceptance-pullrequest(master)/541/functional-upgrade-tcserver40-mysql57-jdk8/2/0/LocalFile.PortalSmokeUpgrade_ViewPortalSmokeArchive625/index.html.gz">Poshi Report</a> - <a href="https://testray.liferay.com/reports/production/logs/test-1-18/1551958464503/test-portal-acceptance-pullrequest(master)/541/functional-upgrade-tcserver40-mysql57-jdk8/2/0/LocalFile.PortalSmokeUpgrade_ViewPortalSmokeArchive625/summary.html.gz">Poshi Summary</a> - <a href="https://testray.liferay.com/reports/production/logs/test-1-18/1551958464503/test-portal-acceptance-pullrequest(master)/541/functional-upgrade-tcserver40-mysql57-jdk8/2/0/jenkins-console.txt.gz">Console Output</a></div></li></ol></div></li></ol></div></li><li><div><h5><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/288042/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tcserver40-mysql57-jdk8/3</a></h5><div><h6>Job Results:</h6><p>0 Tests Passed.<br/>0 Tests Failed.</p></div><ol><li><div><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/288042//consoleText">AXIS_VARIABLE=0,label_exp=!master #288042</a><pre><code>Build was aborted</code></pre></div></li></ol></div></li><li>...</li></ol></details></html>

marianoalvarosaiz commented:
Hi @topolik,
 so, should I send you a pull renaming the private method and changing, inside the API, the position where we use it?
Kind regards.

topolik commented:
Thanks for all the comments @marianoalvarosaiz.

Yes please, I don't have spare time now so I would welcome if you can resend that!

Thanks a lot!

marianoalvarosaiz commented:
see https://github.com/topolik/liferay-portal/pull/722
